pkg/workspace/data_values.go

```diff
@@ -138,45 +151,47 @@ func parseDVAnnotations(doc *yamlmeta.Document) (bool, []LibPathPiece, bool, err
 			afterLibMod, err = core.NewStarlarkValue(kwarg[1]).AsBool()
 			if err != nil {
 				return false, nil, false, err
-			} else if len(libPath) == 0 {
+			} else if len(libRef) == 0 {
 				return false, nil, false, fmt.Errorf("Annotation %s: Expected kwarg 'after_library_module' to be used with %s annotation",
-					yttlibrary.AnnotationDataValues, AnnotationLibraryName)
+					yttlibrary.AnnotationDataValues, AnnotationLibraryRef)
 			}
 		default:
 			return false, nil, false, fmt.Errorf("Unknown kwarg %s for annotation %s", kwargName, yttlibrary.AnnotationDataValues)
 		}
 	}
-	return hasLibAnn, libPath, afterLibMod, nil
+	return hasLibAnn, libRef, afterLibMod, nil
 }
 
-func parseLibPathStr(libPathStr string) ([]LibPathPiece, error) {
-	if libPathStr == "" {
+func parseLibRefStr(libRefStr string) ([]LibRefPiece, error) {
+	if libRefStr == "" {
 		return nil, fmt.Errorf("Expected library name to not be empty")
 	}
-	if !strings.HasPrefix(libPathStr, dvsLibrarySep) {
+
+	if !strings.HasPrefix(libRefStr, dvsLibrarySep) {
 		return nil, fmt.Errorf("Expected library name to start with '%s'", dvsLibrarySep)
 	}
 
-	var result []LibPathPiece
-	for _, libPathPiece := range strings.Split(libPathStr, dvsLibrarySep)[1:] {
-		libAndTag := strings.SplitN(libPathPiece, dvsLibraryTagSep, 2)
-		piece := LibPathPiece{LibName: libAndTag[0]}
-		if len(libAndTag) == 2 {
-			if libAndTag[1] == "" {
-				return nil, fmt.Errorf("Expected library tag to not be empty")
+	var result []LibRefPiece
+	for _, refPiece := range strings.Split(libRefStr, dvsLibrarySep)[1:] {
+		pathAndAlias := strings.Split(refPiece, dvsLibraryAliasIndicator)
+		switch l := len(pathAndAlias); {
+		case l > 2:
+			return nil, fmt.Errorf("Expected library alias not to contain more than one '%s'", dvsLibraryAliasIndicator)
```

Comment:
it's probably worth just stating expected format here eg `Expected library ref piece 'blah' to be in format 'library', '~alias' or 'library~alias'`, instead of trying to describe it in english.